src/evaluate.js

```diff
@@ -14,82 +15,91 @@ export default function evaluate(tokens, expr, values) {
   for (var i = 0; i < numTokens; i++) {
     var item = tokens[i];
     var type = item.type;
+    var token = item.value;
     if (type === INUMBER || type === IVARNAME) {
-      nstack.push(item.value, item.value);
+      nstack.push(type, token);
     } else if (type === IOP2) {
-      var right = nstack.pop();
-      var left = nstack.pop();
-      n2 = right.value;
-      n1 = left.value;
-      if (item.value === 'and') {
-        nstack.push(item.value, n1 ? !!evaluate(n2, expr, values) : false);
-      } else if (item.value === 'or') {
-        nstack.push(item.value, n1 ? true : !!evaluate(n2, expr, values));
-      } else if (item.value === '=') {
-        f = expr.binaryOps[item.value];
-        nstack.push(item.value, f(n1, evaluate(n2, expr, values), values));
-      } else if ((item.value === '+' || item.value === '-') && right.token === '#' && right.token !== left.token) {
-        f = expr.binaryOps[item.value];
+      op2 = nstack.pop();
```

Comment:
quibble: aaah :D i miss the left and right thingy :D to be honest, op1 and n1 look a bit ambiguous, just being nitpicky :)

Reply:
The idea is to keep naming consistency. It is easier to understand that `n1` refers to `op1` than to `left`. And as I don't want to change them too much the code, I kept the other names. Actually, we may need to merge the fixes from the base branch, and a heavy refractory would introduce too many conflicts...